pysm3/models/template.py

```diff
@@ -152,17 +181,29 @@ def apply_smoothing_and_coord_transform(
             hp.smoothalm(alm, fwhm=fwhm.to_value(u.rad), inplace=True, pol=True)
         if rot is not None:
             rot.rotate_alm(alm, inplace=True)
-        smoothed_map = hp.alm2map(alm, nside=nside, pixwin=False)
-
+        if return_healpix:
+            output_maps.append(hp.alm2map(alm, nside=output_nside, pixwin=False))
+        if return_car:
+            shape, wcs = pixell.enmap.fullsky_geometry(
+                output_car_resol.to_value(u.radian)
+            )
+            ainfo = pixell.sharp.alm_info(lmax=lmax)
+            output_maps.append(
+                pixell.curvedsky.alm2map(
+                    alm, pixell.enmap.empty(shape, wcs), ainfo=ainfo
+                )
+            )
```

Comment:
@msyriac I have the alms and would like to generate both HEALPix and CAR.
I implemented this using curvedsky.alm2map.

But in the unit test below when I compare it to enmap_from_healpix, the unit test fails badly.
Can you check if I am doing something wrong? I am worried there are different assumptions in the 2 methods

Reply:
@msyriac if there is nothing obvious just looking at the code I can provide a standalone notebook

Reply:
Sigurd helped me out on Slack

Reply:
it was the different transform healpix to alm in pixell and in healpy